python/TestHarness/schedulers/Scheduler.py

```diff
@@ -0,0 +1,539 @@
+        closes the status pool to prevent further statuses from printing to the
+        screen.
+        """
+        self.tester_pool.close()
+        self.status_pool.close()
+
+        for tester_data in self.tester_datas:
+            tester_data.killProcess()
+        self.job_queue_count = 0
+
+    def run(self, job_container):
+        """ call derived run method """
+        return
+
+    def skipPrereqs(self):
+        """
+        Method to return boolean to skip dependency prerequisites checks.
+        """
+        if self.options.ignored_caveats:
+            if 'all' in self.options.ignored_caveats or 'prereq' in self.options.ignored_caveats:
+                return True
+        return False
+
+    def skippedTests(self, testers):
+        """
+        Method to return a list of skipped tests based on their current status and
+        their caveat check (getRunnable).
+        """
+        skipped_failed = set([])
+        for tester in testers:
+            if (tester.isFinished() and not tester.didPass()) \
+               or not tester.getRunnable(self.options):
+                skipped_failed.add(tester)
+        return skipped_failed
+
+    def processDownstreamTests(self, job_container):
+        """
+        Method to discover and delete downstream jobs due to supplied job failing.
+        """
+        with self.dag_lock:
+            failed_job_containers = set([])
+            tester = job_container.getTester()
+            job_dag = job_container.getDAG()
+            if (tester.isFinished() and not tester.didPass() and not tester.isSilent() and not self.skipPrereqs()) \
+                or (self.options.dry_run and not tester.isSilent()):
+
+                # Ask the DAG to delete and return the downstream jobs associated with this job
+                failed_job_containers.update(job_dag.delete_downstreams(job_container))
+
+            for failed_job in failed_job_containers:
+                tester = failed_job.getTester()
+                tester.setStatus('skipped dependency', tester.bucket_skip)
+
+        return failed_job_containers
+
+    def createDependencies(self, job_container_dict):
+        """
+        Create dependencies using supplied dictionary of TesterData objects.
+        """
+        for tester_name, job_container in job_container_dict.iteritems():
+            tester = job_container.getTester()
+            job_dag = job_container.getDAG()
+
+            job_dag.add_node_if_not_exists(job_container)
+
+            for prereq in tester.getPrereqs():
+
+                if prereq in job_container_dict.keys():
+
+                    job_dag.add_node_if_not_exists(job_container_dict[prereq])
+
+                    # Attempt to create an edge node dependency
+                    try:
+                        job_dag.add_edge(job_container_dict[prereq], job_container)
+
+                    # We do not care about cyclic errors, they have been handled already
```

Comment:
Hmm, I argue that you do care about them here. As you said, they have been handled already so if you were to hit one, something very bad and unexpected would have happened. I suggest that you allow this to crash because if it ever does - we'll have to investigate.